source/text/Format.cpp

```diff
@@ -27,7 +27,9 @@ this program. If not, see <https://www.gnu.org/licenses/>.
 using namespace std;
 
 namespace {
-	const int64_t K = 1000;
+	// The greatest number displayed without switching to scientific notation.
+	constexpr int64_t SCIENTIFIC_THRESHOLD = 1'000'000'000'000'000ll;
```

Comment:
```suggestion
	constexpr int64_t SCIENTIFIC_THRESHOLD = 999'999'999'999'999ll;
```
Otherwise it might overflow of something has exactly 1 quintillion

Reply:
Nah the overflow would happen later, but sure.